crates/biome_js_analyze/src/analyzers/nursery/no_implicit_any_let.rs

```diff
@@ -0,0 +1,90 @@
+use biome_analyze::{context::RuleContext, declare_rule, Ast, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::{JsFileSource, JsVariableDeclaration, JsVariableDeclarator};
+
+declare_rule! {
+    /// Restrict use of implicit any type in Typescript.
+    ///
+    /// Typescript variable declaration without any `type` or `initialization` can cause issue later in the code.
+    ///
+    ///
+    ///
+    /// Source: https://www.typescriptlang.org/tsconfig#noImplicitAny
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// var a;
+    /// a = 2;
+    /// let b;
+    /// b = 1
```

Comment:
Each snippet in the **invalid section** must raise only **one** diagnostic, this means that you'd need to break down this snippet in **four** code blocks with `expect_diagnostic` directive.

The valid section doesn't have this restriction.